### DISCUSSION TITLE
Cargo Install Error - `build.rs: No such file or directory`

### DISCUSSION THREAD

**Discussion by fatsanta92** [ReagentX/imessage-exporter#417] [Q&A] [answered]
I keep trying to install via cargo and get two error messages:

error: couldn't read /Users/UserName/.cargo/registry/src/index.crates.io-6f17d22bba15001f/imessage-database-2.2.2/build.rs: No such file or directory (os error 2)

error: could not compile `imessage-database` (build script) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `imessage-exporter v2.2.2`, intermediate artifacts can be found at `/var/folders/zf/qzw0rrbs04jg_2yt00b03gf80000gn/T/cargo-installWghVgq`.

**Comment by ReagentX** [comment]
What platform are you using and how did you install `imessage-exporter`? It seems that `cargo` didn't download the `build.rs` file for some reason, but it definitely [should be included](https://docs.rs/crate/imessage-database/2.2.2/source/build.rs).

**Reply by fatsanta92** [reply, marked as answer]
I'm using Mac OS Sequoia 15.1.1, I installed using terminal + the cargo install folder.

I ended up downloading the file directly from github and dropping it into the correct location and re-running the command which worked. 

Figured I'd bring it to your attention but maybe it's just a me issue.

**Reply by ReagentX** [reply]
I was able to run `cargo install imessage-exporter` Using `cargo 1.83.0`  macOS `15.2 (24C101)` just now.
